src/hipscat/io/file_io/file_pointer.py

```diff
@@ -133,105 +23,94 @@ def append_paths_to_pointer(pointer: FilePointer, *paths: str) -> FilePointer:
     Returns:
         New file pointer to path given by joining given pointer and path names
     """
-    return FilePointer(os.path.join(pointer, *paths))
+    pointer = get_upath(pointer)
+    return pointer.joinpath(*paths)
```

Comment:
`get_upath` can return None, and this would then raise an error. That might be ok behavior though, but might also want to give a more helpful error message.